code/game/objects/items/devices/atomic_disassembler.dm

```diff
@@ -0,0 +1,156 @@
+/obj/item/device/atomic_disassembler
+	name = "P.A.D.D."
+	desc = "The Portable Atomic Disassembly Device (patent pending) is a revolutionary device in breaking down objects at the atomic level. To use, you must attach a condensed matter canister and a power \
+	cell. Then, simply feed an object through its atomization field. It will then be broken down into raw condensed matter and placed into the canister, which can be utilized in many ways!"
+	hitsound = 'sound/weapons/genhit1.ogg'
+	w_class = 3 //Maybe make bulky sometime?
+	force = 4
+	slot_flags = SLOT_BACK
+	origin_tech = "programming=4;materials=4;magnets=4;bluespace=4"
+	throw_range = 3
+	throw_speed = 1
+	icon_state = "atom_inactive"
+	mouse_opacity = 2
+	var/obj/item/weapon/stock_parts/cell/power_cell = null
+	var/obj/item/device/compressed_matter_canister/canister = null
```

Comment:
You need to clear these references in Destroy() either by deleting them or just setting the reference to null and ejecting them outside.